subprojects/gradle/slack/src/integTest/kotlin/com/avito/slack/SlackConditionalSenderIntegrationTest.kt

```diff
@@ -87,4 +97,9 @@ internal class SlackConditionalSenderIntegrationTest {
             emoji = ":crazy-robot:"
         )
     ).onFailure { it.printStackTrace() }
+
+    private fun String?.assumeNotNullOrBlank(): String {
+        Assumptions.assumeFalse { isNullOrBlank() }
```

Comment:
It looks like a half measure.
What are we fixing? Local checks?

1. Such optimizations give a false sense of safety in local runs.
2. We can't do a fail-fast check in CI. 
Imagine a situation when we lose these ENVs in CI. Tests are green but we expected a failure.

Let's choose a more explicit way.

Reply:
any suggestions?

Reply:
tests will be ignored, not green.
We can forbid ignored tests at all in CI

Reply:
> We can forbid ignored tests at all in CI

In this case, we need to distinguish explicitly `@Disabled` tests from implicitly skipped by assumptions.

I propose another solution, to pass an intention through parameters, e.g.:

- Default value in CI properties: `SLACK_INTEGRATION_TESTS` == enabled -> requires other parameters (url, token, ...).
- Default value in the project: `SLACK_INTEGRATION_TESTS` == disabled -> skip tests by assumptions
- Else (unset or unknown value) -> fail fast, require to pass the intention. 

We use a similar pattern in gradle plugins. 
It protects us from accidentally lost parameters. 

Also, I like to separate concerns when feature enabled and not configured properly.
If I don't have SLACK_TEST_CHANNEL it shouldn't mean that I don't want to run tests. It's harder to understand and triage.